packages/nimble-components/src/drawer/index.ts

```diff
@@ -131,6 +131,12 @@ export class Drawer<CloseReason = void> extends FoundationElement {
     }
 
     private triggerAnimation(): void {
+        // Read the offsetHeight of the dialog to trigger a reflow. This guarantees that the browser
+        // has processed the 'animating' class being removed before trying to readd it, even if the previous
+        // animation has just finished. Otherwise, problems can occur. For example, trying to close the
+        // drawer immediately after the opening animation ends does not actually close the drawer.
+        void this.dialog.offsetHeight;
```

Comment:
Moving comment to discussion:

>> The only other thought I had was that we could explore using the (JavaScript) Web Animations API instead of the current CSS animation approach. But I took a look at the dialog styles and that's probably not feasible due to the animations on the backdrop (I'm not aware of a JS equivalent to the ::backdrop pseudo-selector).

>There is a pseudoElement option for element.animate: https://caniuse.com/mdn-api_element_animate_options_pseudoelement_parameter
> 
>Haven't tried it with ::backdrop specifically

Reply:
I tried using the javascript web animations API as suggested, but it turns out the `::backdrop` psuedo element isn't supported in FF. https://bugzilla.mozilla.org/show_bug.cgi?id=1770591

FYI - @rajsite

Reply:
Ahh dang, lovely. FWIW this isn't the first time this is being done for FF in the monorepo:

https://github.com/ni/nimble/blob/8a0315cd3a2789f38ad69d8fbeab0d88c085b253/packages/site/landing/circle.ts#L14-L16

Can't think of much else to do. Not sure if it's worth tracking as a tech debt item to switch to web animations when possible. Maybe doesn't hurt to capture the details in an item and link from here.